### ISSUE TITLE
Multiple Interfaces Created in Container

### ISSUE THREAD

**Issue by Kent-H** [contiv/netplugin#872]
I was poking around with the demo-v2plugin, and discovered that every container created has two network interfaces.

I can't seem to create a container with *only* the contiv network, there is always another interface which provides internet connectivity.  I want to create containers which are connected directly to (one or more) contiv networks, and nothing else.

I would have expected this to be the default behavior.  Is this a bug, or intended functionality?

**Comment by gkvijay** [comment]
@Kent-H: When you created the contiv network, was the gateway option also specified?

**Comment by Kent-H** [comment]
I don't think it was...  Are subnet & gateway both required?  (My eventual goal is to have networks with no subnet, gateway, or IP addressing of any kind, in order to run Layer 2 services.)

**Comment by gkvijay** [comment]
The subnet is always required (L2 or L3). The IP address for the container is allocated from this pool. The gateway is optional in L2. If contiv network is configured with a gateway, traffic to a destination outside the network subnet will be sent to gateway mac for routing. If contiv network is created without a gateway, then Docker creates an additional interface to connect the container to docker bridge for traffic going outside the cluster. In this traffic going outside the network is not controlled by contiv.

**Comment by Kent-H** [comment]
Why is a subnet/IP assigned, even in L2 mode?  IP is usually considered L3...

Having a subnet & IP is currently a requirement for docker network plugins.  Perhaps this is why contiv requires them?

**Comment by gkvijay** [comment]
In L2-mode forwarding in contiv will use mac address lookup, but the containers need to have IP address. Applications will use IP address of container independent of the networking mode (L2/L3).
